modules/oe_content_organisation/oe_content_organisation.module

```diff
@@ -0,0 +1,127 @@
+<?php
+
+/**
+ * @file
+ * OE Content Organisation module file.
+ */
+
+declare(strict_types = 1);
+
+use Drupal\Core\Entity\EntityInterface;
+use Drupal\Core\Entity\EntityTypeInterface;
+use Drupal\Core\Field\BaseFieldDefinition;
+use Drupal\Core\Field\FieldStorageDefinitionInterface;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\rdf_skos\Entity\Concept;
+use Drupal\rdf_skos\Entity\ConceptInterface;
+
+/**
+ * Implements hook_form_BASE_FORM_ID_alter for the node form.
+ */
+function oe_content_organisation_form_node_form_alter(array &$form, FormStateInterface $form_state) {
+  /** @var \Drupal\node\NodeInterface $node */
+  $node = $form_state->getBuildInfo()['callback_object']->getEntity();
+  if ($node->bundle() !== 'oe_organisation') {
+    return;
+  }
+
+  $form['oe_organisation_non_eu_org_type']['widget'][0]['target_id']['#states'] = [
+    'visible' => [
+      ['select[name="oe_organisation_org_type"]' => ['value' => 'non_eu']]
+    ],
+    'required' => [
+      ['select[name="oe_organisation_org_type"]' => ['value' => 'non_eu']]
+    ]
+  ];
+
+  $form['oe_organisation_eu_org']['widget'][0]['target_id']['#states'] = [
+    'visible' => [
+      ['select[name="oe_organisation_org_type"]' => ['value' => 'eu']]
+    ],
+    'required' => [
+      ['select[name="oe_organisation_org_type"]' => ['value' => 'eu']]
+    ]
+  ];
+
+  $form['#validate'][] = 'oe_content_organisation_node_validate';
+}
+
+/**
+ * Custom validator for the Organisation node form.
+ *
+ * @param array $form
+ *   The form.
+ * @param \Drupal\Core\Form\FormStateInterface $form_state
+ *   The form state.
+ */
+function oe_content_organisation_node_validate(array $form, FormStateInterface $form_state): void{
+  // Ensure that organisation values are correctly selected.
+  $organisation_type = $form_state->getValue(['oe_organisation_org_type', 0, 'value']);
+  $non_eu_organisation_type = $form_state->getValue(['oe_organisation_non_eu_org_type', 0, 'target_id']);
+  $eu_organisation = $form_state->getValue(['oe_organisation_eu_org', 0, 'target_id']);
+  if ($organisation_type === 'eu' && !$eu_organisation) {
+    $form_state->setErrorByName('oe_organisation_eu_org', t('Please select an EU organisation.'));
```

Comment:
This is not tested in your Behat scenario. There should be a step that removes the browser-side "required" validation so you can submit the form and test this as well.